config/src/test/java/org/springframework/security/config/http/OAuth2LoginBeanDefinitionParserTests.java

```diff
@@ -0,0 +1,739 @@
+
+		Authentication authentication = this.securityContextRepository
+				.loadContext(new HttpRequestResponseHolder(request, response)).getAuthentication();
+		assertThat(authentication.getAuthorities()).hasSize(1);
+		assertThat(authentication.getAuthorities()).first().isInstanceOf(OAuth2UserAuthority.class)
+				.hasToString("ROLE_USER");
+	}
+
+	// gh-6009
+	@Test
+	public void successLoginWhenSingleClientRegistrationThenAuthenticationSuccessEventPublished() throws Exception {
+		this.spring.configLocations(this.xml("SingleClientRegistration-WithTestConfiguration")).autowire();
+
+		MockHttpSession session = new MockHttpSession();
+		MockHttpServletRequest request = new MockHttpServletRequest("GET", "");
+		request.setServletPath("/login/oauth2/code/google");
+		request.setSession(session);
+		MockHttpServletResponse response = new MockHttpServletResponse();
+
+		OAuth2AuthorizationRequest authRequest = createOAuth2AuthorizationRequest(
+				clientRegistrationRepository.findByRegistrationId("google-login"));
+		authorizationRequestRepository.saveAuthorizationRequest(authRequest, request, response);
+
+		MultiValueMap<String, String> params = new LinkedMultiValueMap<>();
+		params.add("code", "code123");
+		params.add("state", "state123");
+		this.mvc.perform(get("/login/oauth2/code/google").params(params).session(session));
+
+		// assertions
+		assertThat(authenticationSuccessListener.events).isNotEmpty();
+		assertThat(authenticationSuccessListener.events).hasSize(1);
+		assertThat(authenticationSuccessListener.events.get(0)).isInstanceOf(AuthenticationSuccessEvent.class);
+	}
+
+	@Test
+	public void successOidcLoginWhenSingleClientRegistrationThenRedirectToDefaultUrl() throws Exception {
+		this.spring.configLocations(this.xml("SingleClientRegistration-WithTestConfiguration")).autowire();
+		// set the jwt decoder to test instance
+		oidcAuthorizationCodeAuthenticationProvider.setJwtDecoderFactory(new MockOAuth2OidcJwtDecoderFactory());
+
+		MockHttpSession session = new MockHttpSession();
+		MockHttpServletRequest request = new MockHttpServletRequest("GET", "");
+		request.setServletPath("/login/oauth2/code/google");
+		request.setSession(session);
+		MockHttpServletResponse response = new MockHttpServletResponse();
+
+		OAuth2AuthorizationRequest authRequest = createOAuth2AuthorizationRequest(
+				clientRegistrationRepository.findByRegistrationId("google-login"), "openid");
+		authorizationRequestRepository.saveAuthorizationRequest(authRequest, request, response);
+
+		MultiValueMap<String, String> params = new LinkedMultiValueMap<>();
+		params.add("code", "code123");
+		params.add("state", "state123");
+		this.mvc.perform(get("/login/oauth2/code/google").params(params).session(session))
+				.andExpect(status().is3xxRedirection()).andExpect(redirectedUrl("/"));
+	}
+
+	@Test
+	public void failedOidcLoginWhenSingleClientRegistrationThenRedirectToDefaultLoginError() throws Exception {
```

Comment:
after i update my test class and think about this again, I plan to remove this, the test case doesn't make sense for me. 

i actually plan to test if `OidcAuthenticationRequestChecker` was triggered but i dont think i can test this because of the logic at `OAuth2LoginBeanDefinitionParser`
		`boolean oidcAuthenticationProviderEnabled = ClassUtils
				.isPresent("org.springframework.security.oauth2.jwt.JwtDecoder", this.getClass().getClassLoader());`

in which during test will always return `true`